src/pages/SystemAdmin/NamespaceCard.tsx

```diff
@@ -17,41 +19,82 @@ import {
 import SystemCard from 'pages/SystemAdmin/SystemAdminCard'
 import { useEffect, useState } from 'react'
 import { System } from 'types/backend-types'
+import { SnackbarState } from 'types/custom-types'
 
 const NamespaceCard = ({ namespace }: { namespace: string }) => {
   const { hasSystemPermission } = PermissionsContainer.useContainer()
-  const systemClient = useSystems()
-  const systems = systemClient.systems
+  const { getSystems } = useSystems()
   const [expanded, setExpanded] = useState(true)
   const [permission, setPermission] = useState(false)
+  const [filteredSystems, setFilteredSystems] = useState<System[]>([])
+  const [sortedSystems, setSortedSystems] = useState<{
+    [key: string]: System[]
+  }>({})
+  const [alert, setAlert] = useState<SnackbarState | undefined>(undefined)
+  const { addCallback, removeCallback } = SocketContainer.useContainer()
 
-  const handleExpandClick = () => {
-    setExpanded(!expanded)
+  const fetchPermissions = (systems: System[]) => {
+    if (systems[0]) {
+      hasSystemPermission('system:update', namespace, systems[0].id).then(
+        (response) => {
+          setPermission(response || false)
+        },
+      )
+    }
   }
 
-  const filteredSystem = systems.filter(
-    (system: System) => system.namespace === namespace,
-  )
-  useEffect(() => {
-    const fetchPermission = async () => {
-      if (filteredSystem[0]) {
-        const permCheck = await hasSystemPermission(
-          'system:update',
-          namespace,
-          filteredSystem[0].id,
+  const updateSystems = () => {
+    getSystems()
+      .then((response) => {
+        return response.data
+      })
+      .then((systems) => {
+        const filtered = systems.filter(
+          (system: System) => system.namespace === namespace,
         )
-        setPermission(permCheck || false)
+        setFilteredSystems(filtered)
+        return filtered
+      })
+      .then((filtered) => {
+        fetchPermissions(filtered)
+        return filtered
+      })
+      .then((filtered) => {
```

Comment:
I did it like this because the fetchPermissions would fail on first render since setFilteredSystems is async and wasn't immediately available. Probably could have been done a bit neater though lol